contracts/forms.py

```diff
@@ -0,0 +1,19 @@
+
+from django import forms
+
+from .models import TenancyContract
+
+
+class DateInput(forms.DateInput):
+    input_type = 'date'
+
+
+class TenancyContractForm(forms.ModelForm):
+
+    class Meta:
```

Comment:
You can reduce the data errors by custom error handling in clean method, for example
```
def clean(self):
        cleaned_data = super().clean()
        start_date = cleaned_data.get("start_date")
        end_date = cleaned_data.get("end_date")

        if end_date and start_date and end_date < start_date:
            raise forms.ValidationError("End date should be greater than start date.")

        return cleaned_data
```